components/datetime/src/neo.rs

```diff
@@ -1097,3 +1264,44 @@ impl FormattedDateTime<'_> {
         self.pattern.to_pattern()
     }
 }
+
+/// An intermediate type during a datetime formatting operation with dynamic input.
+///
+/// Unlike [`FormattedDateTime`], converting this to a string could fail.
+///
+/// Not intended to be stored: convert to a string first.
+#[derive(Debug)]
+pub struct FormattedDateTimeTry<'a> {
```

Comment:
Happy to bikeshed `FormattedDateTimeTry`

Reply:
FallibleFormattedDateTime probably.

Reply:
So we have `FormattedDateTime` and `FormattedDateTimePattern`, so it seems nice to keep at least `Formatted` as the prefix for sorting purposes. I'd be okay with:

- `FormattedDateTimeFallible`
- `FormattedDateTimeUnchecked`
- `FormattedDateTimeChecked`

Reply:
FormattedDateTimeUnchecked